components/automate-ui/src/app/modules/infra-proxy/cookbook-details/cookbook-details.component.spec.ts

```diff
@@ -0,0 +1,149 @@
+import { CookbookDetailsComponent } from './cookbook-details.component';
+
+
+const declarations: any[] = [
+  MockComponent({ selector: 'chef-heading' }),
+  MockComponent({ selector: 'chef-icon' }),
+  MockComponent({ selector: 'chef-loading-spinner' }),
+  MockComponent({ selector: 'mat-select' }),
+  MockComponent({ selector: 'mat-option' }),
+  MockComponent({ selector: 'chef-page-header' }),
+  MockComponent({ selector: 'chef-subheading' }),
+  MockComponent({ selector: 'chef-toolbar' }),
+  MockComponent({ selector: 'a', inputs: ['routerLink'] }),
+  MockComponent({ selector: 'input', inputs: ['resetOrigin'] }),
+  CookbookDetailsComponent
+];
+const serverId = '6e98f609-586d-4816-a6de-e841e659b11d';
+const orgId = '6e98f609-586d-4816-a6de';
+const cookbook_name = 'aix';
+const cookbookVersion: CookbookVersions = {
+  name: 'cookbook version',
+  versions: []
+};
+
+const cookbookDetails: CookbookDetails = {
+    cookbook_name: 'aix',
+    name_and_version: 'aix_1.1.1',
+    version: '1.1.1',
+    chef_type: 'any',
+    frozen: 'any',
+    json_class: '1',
+    files: [],
+    templates: [],
+    attributes: [],
+    recipes: [],
+    definitions: [],
+    libraries: [],
+    providers: [],
+    resources: [],
+    root_files: [],
+    metadata: [],
+    access: []
+  };
+
+describe('CookbookDetailsComponent', () => {
+  let component: CookbookDetailsComponent;
+  let fixture: ComponentFixture<CookbookDetailsComponent>;
+  let router: Router;
+  let store: Store<NgrxStateAtom>;
+
+  const initialState = {
+    ...defaultInitialState,
+    router: {
+      ...defaultRouterState,
+      state: {
+        ...defaultRouterRouterState,
+        url: `infrastructure/chef-servers/${serverId}/org/${orgId}/cookbooks/${cookbook_name}`,
+        params: { id: serverId, orgid: orgId, cookbook_name: cookbook_name }
+      }
+    }
+  };
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: declarations,
+      providers: [
+        FeatureFlagsService
+      ],
+      imports: [
+        RouterTestingModule,
+        HttpClientTestingModule,
+        StoreModule.forRoot(ngrxReducers, { initialState, runtimeChecks })
+      ],
+      schemas: [ CUSTOM_ELEMENTS_SCHEMA ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    router = TestBed.inject(Router);
+    spyOn(router, 'navigate').and.stub();
+    store = TestBed.inject(Store);
+
+    fixture = TestBed.createComponent(CookbookDetailsComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should be created', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('defaults to showing users section', () => {
+    expect(component.tabValue).toBe('details');
+  });
+
+  describe('empty state', () => {
+    beforeEach(() => {
+      store.dispatch(new GetCookbookVersionsSuccess(cookbookVersion));
+      fixture.detectChanges();
+    });
+
+    it('cookbook should be empty', () => {
```

Comment:
I do not think that this test really adds value, so you might remove it. Checking solely that something is undefined would likely be true when you start most chunks of code.